### ISSUE TITLE
Cifar10 example sometimes only predicts 1 class and does not improve

### ISSUE THREAD

**Issue by tslevi-unbounce** [fchollet/keras#4066]
Running this script

https://github.com/fchollet/keras/blob/master/examples/cifar10_cnn.py

it sometimes locks into a situation where it only predicts a single class and accuracy fixes at 0.1 .

**Comment by tslevi** [comment]
[cifar10.pdf](https://github.com/fchollet/keras/files/530603/cifar10.pdf)

**Comment by aidiary** [comment]
I also encountered the same problem before.
I solved using Adam instead of SGD.

**Comment by NYcleaner** [comment]
@aidiary  yes,  the adam optimizer works fine , i just try it

**Comment by wangdelp** [comment]
@aidiary Hi, I've changed to Adam but the loss still get stucked at 2.304, and accu=0.1.
are you using Adam with lr 0.01? Is there any other configs for Adam? Thank you.

**Comment by aidiary** [comment]
@wangdelp No, I am using Adam with default settings. Here is my sample codes.
https://github.com/aidiary/keras_examples/blob/master/cnn/cifar10/cifar10.py
```python
model.compile(loss='categorical_crossentropy',
                  optimizer='adam',
                  metrics=['accuracy'])
```

**Comment by wangdelp** [comment]
@aidiary Thank you, by changing the optimizer to Adam without any further config is by default using a smaller learning rate 0.001 as far as I can see from the Adam class #L315 in the file:
https://github.com/fchollet/keras/blob/master/keras/optimizers.py. So I thought the difference should still comes with the different learning rate used. Thanks.
